src/components/datepicker/js/datepickerDirective.js

```diff
@@ -402,9 +402,17 @@
 
     // Responds to external changes to the model value.
     self.ngModelCtrl.$formatters.push(function(value) {
+      var parsedValue = angular.isDefined(value) ? Date.parse(value) : null;
+
+      if (parsedValue) {
+        value = self.dateLocale.parseDate(parsedValue);
```

Comment:
The difference between `Date.parse` and `new Date` is that `parse` is a little more strict about what it accepts. E.g. the constructor will parse `null`, `false`, etc. into January 1st 1970, whereas `parse` with consider them invalid. Also `parse` returns the time since epoch, which is why it needs to be passed through `parseDate`.

As for the reason why it's not doing the logic in `parseDate`, it's because this should only be done when something, different from the directive, modifies the model.